source/dynamical_systems/include/dynamical_systems/DefaultDynamicalSystem.hpp

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include "IDynamicalSystem.hpp"
+
+namespace dynamical_systems {
+
+/**
+ * @class DefaultDynamicalSystem
+ * @brief A default dynamical system that just returns an empty state.
+ * @tparam S Underlying state type of the dynamical system
+ */
+template<class S>
+class DefaultDynamicalSystem : public IDynamicalSystem<S> {
+public:
+  /**
+   * @brief Compute the dynamics of the input state.
+   * @param state The input state
+   * @return The output state
+   */
+  [[nodiscard]] S compute_dynamics(const S& state) const override;
```

Comment:
I guess this makes sense to declare the `const` function as `nodiscard`. There is a lot of those we didn't do yet though. Should we make it the new pattern from now on?

Reply:
My opinion that I discussed with dominic is that we should use this marking when convenient, since it is otherwise a soft warning in C++17, but I don't mind if it is absent in some places (so it's not something that necessitates a refactor, but for new code we might as well)